test/test_distributions.py

```diff
@@ -480,6 +488,63 @@ def test_gamma_sample_grad(self):
                                        'rel error {}'.format(rel_error),
                                        'max error {}'.format(rel_error.max())]))
 
+    # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_chi2_shape(self):
+        df = Variable(torch.exp(torch.randn(2, 3)), requires_grad=True)
+        df_1d = Variable(torch.exp(torch.randn(1)), requires_grad=True)
+        self.assertEqual(Chi2(df).sample().size(), (2, 3))
+        self.assertEqual(Chi2(df).sample_n(5).size(), (5, 2, 3))
+        self.assertEqual(Chi2(df_1d).sample_n(1).size(), (1, 1))
+        self.assertEqual(Chi2(df_1d).sample().size(), (1,))
+        self.assertEqual(Chi2(0.5).sample().size(), (1,))
+        self.assertEqual(Chi2(0.5).sample_n(1).size(), (1,))
+
+        def ref_log_prob(idx, x, log_prob):
+            d = df.data.view(-1)[idx]
+            expected = scipy.stats.chi2.logpdf(x, d)
+            self.assertAlmostEqual(log_prob, expected, places=3)
+
+        self._check_log_prob(Chi2(df), ref_log_prob)
+
+    # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_chi2_sample(self):
+        self._set_rng_seed()
+        for df in [0.1, 1.0, 5.0]:
+            self._check_sampler_sampler(Chi2(df),
+                                        scipy.stats.chi2(df),
+                                        'Chi2(df={})'.format(df))
+
+        # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_chi2_sample_grad(self):
+        self._set_rng_seed(1)
+        num_samples = 100
+        for df in [1e-3, 1e-1, 1e0, 1e1, 1e2, 1e3, 1e4]:
+            # NOTE: for some reason test fails for df = 1e-2
```

Comment:
I've created an issue for this and @martinjankowiak and I will see if we can fix it in the next couple weeks https://github.com/probtorch/pytorch/issues/48